ql/src/java/org/apache/hadoop/hive/ql/parse/type/RexNodeExprFactory.java

```diff
@@ -613,6 +617,14 @@ protected RexNode createConstantExpr(TypeInfo typeInfo, Object constantValue)
         TypeConverter.convert(typeInfo, rexBuilder.getTypeFactory()), false);
   }
 
+  /**
+   * Special operator that is used as syntactic sugar to change the type of collection
+   * expressions in order to perform field access over them.
+   */
+  public static final SqlOperator COMPONENT_ACCESS =
```

Comment:
Most of the existing Hive specific operators/functions are under `org.apache.hadoop.hive.ql.optimizer.calcite.reloperators` package (e.g., `HiveDateAddSqlOperator`). Consider moving the new operator into a separate class under that package.

Having one class per function is not ideal but it may be a good idea to follow the existing paradigm at least till we decide to perform a holistic refactoring that puts them all together.